pkg/bindinfo/tests/bind_test.go

```diff
@@ -880,3 +881,232 @@ func TestNormalizeStmtForBinding(t *testing.T) {
+			`select * from t1, %st2, t3 where t1.a=t2.a and t3.b=t2.b`},
+		{`create global binding using select /*+ no_merge_join(t1) */ * from *.t1, *.t2 where t1.b=t2.b`,
+			`select * from %st1, t2 where t1.b=t2.b`},
+		{`create global binding using select /*+ no_merge_join(t2) */ * from *.t1, *.t2 where t1.c=t2.c`,
+			`select * from t1, %st2 where t1.c=t2.c`},
+		{`create global binding using select /*+ no_merge_join(t2) */ * from *.t1, *.t2, *.t3 where t1.a=t2.a and t3.b=t2.b`,
+			`select * from t1, %st2, t3 where t1.a=t2.a and t3.b=t2.b`},
+		{`create global binding using select /*+ no_index_join(t1) */ * from *.t1, *.t2 where t1.b=t2.b`,
+			`select * from %st1, t2 where t1.b=t2.b`},
+		{`create global binding using select /*+ no_index_join(t2) */ * from *.t1, *.t2 where t1.c=t2.c`,
+			`select * from t1, %st2 where t1.c=t2.c`},
+		{`create global binding using select /*+ no_index_join(t2) */ * from *.t1, *.t2, *.t3 where t1.a=t2.a and t3.b=t2.b`,
+			`select * from t1, %st2, t3 where t1.a=t2.a and t3.b=t2.b`},
+
+		// join order hint
+		{`create global binding using select /*+ leading(t2) */ * from *.t1, *.t2 where t1.b=t2.b`,
+			`select * from %st1, t2 where t1.b=t2.b`},
+		{`create global binding using select /*+ leading(t2) */ * from *.t1, *.t2 where t1.c=t2.c`,
+			`select * from t1, %st2 where t1.c=t2.c`},
+		{`create global binding using select /*+ leading(t2, t1) */ * from *.t1, *.t2 where t1.c=t2.c`,
+			`select * from t1, %st2 where t1.c=t2.c`},
+		{`create global binding using select /*+ leading(t1, t2) */ * from *.t1, *.t2 where t1.c=t2.c`,
+			`select * from t1, %st2 where t1.c=t2.c`},
+		{`create global binding using select /*+ leading(t1) */ * from *.t1, *.t2, *.t3 where t1.a=t2.a and t3.b=t2.b`,
+			`select * from t1, %st2, t3 where t1.a=t2.a and t3.b=t2.b`},
+		{`create global binding using select /*+ leading(t2) */ * from *.t1, *.t2, *.t3 where t1.a=t2.a and t3.b=t2.b`,
+			`select * from t1, %st2, t3 where t1.a=t2.a and t3.b=t2.b`},
+		{`create global binding using select /*+ leading(t2,t3) */ * from *.t1, *.t2, *.t3 where t1.a=t2.a and t3.b=t2.b`,
+			`select * from t1, %st2, t3 where t1.a=t2.a and t3.b=t2.b`},
+		{`create global binding using select /*+ leading(t2,t3,t1) */ * from *.t1, *.t2, *.t3 where t1.a=t2.a and t3.b=t2.b`,
+			`select * from t1, %st2, t3 where t1.a=t2.a and t3.b=t2.b`},
+	} {
+		removeAllBindings(tk, true)
+		tk.MustExec(c.binding)
+		for _, currentDB := range []string{"db1", "db2", "db3"} {
+			tk.MustExec(`use ` + currentDB)
+			for _, db := range []string{"db1.", "db2.", "db3.", ""} {
+				query := fmt.Sprintf(c.qTemplate, db)
+				tk.MustExec(query)
+				tk.MustQuery(`show warnings`).Check(testkit.Rows()) // no warning
+				tk.MustExec(query)
+				tk.MustQuery(`select @@last_plan_from_binding`).Check(testkit.Rows("1"))
+			}
+		}
+	}
+}
+
+func TestFuzzyBindingHints(t *testing.T) {
+	testFuzzyBindingHints(t)
+}
+
+func TestFuzzyBindingHintsWithSourceReturning(t *testing.T) {
```

Comment:
Please add one more test case to ensure we won't return any error to the user if the storage loading process is wrong, and the optimization can continue.

Reply:
I will add these test cases in the next PR.